src/core/AutomationPattern.cpp

```diff
@@ -279,23 +312,37 @@ void AutomationPattern::recordValue(MidiTime time, float value)
 
 /**
  * @brief Set the position of the point that is being dragged.
- *        Calling this function will also automatically set m_dragging to true,
- *        which applyDragValue() have to be called to m_dragging.
- * @param the time(x position) of the point being dragged
- * @param the value(y position) of the point being dragged
- * @param true to snip x position
- * @return
+ *        Calling this function will also automatically set m_dragging to true.
+ *        When applyDragValue() is called to m_dragging is set back to false.
+ * @param MidiTime of the node being dragged
+ * @param Float with the value to assign to the point being dragged
+ * @param Boolean. True to snip x position
+ * @param Boolean. True to ignore unquantized surrounding nodes
+ * @return MidiTime with current time of the dragged value
  */
 MidiTime AutomationPattern::setDragValue( const MidiTime & time,
 						const float value,
 						const bool quantPos,
 						const bool controlKey )
 {
+	QMutexLocker m(&m_patternMutex);
+
 	if( m_dragging == false )
 	{
 		MidiTime newTime = quantPos  ?
 				Note::quantized( time, quantization() ) :
 							time;
+
+		m_dragOutValueOffset = 0;
+
+		// Check if we already have a node on the position we are dragging
```

Comment:
I'm not a fan of this behavior, when I drag an in point I expect the out point to stay where it is.